security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -335,6 +323,52 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+func (sc *SecretCache) addFileWatcher(file string, token string, connKey ConnKey) {
+	// TODO(ramaraochavali): add integration test for file watcher functionality.
+	// Check if this file is being already watched, if so ignore it. FileWatcher has the functionality of
+	// checking for duplicates. This check is needed here to avoid processing duplicate events for the same file.
+	sc.certMutex.Lock()
+	_, exists := sc.fileCerts[file]
```

Comment:
1. Can you make sure this path is normalized so that different path names that resolving to the same file can be treated as the same one.
2. What about multiple connections are referring to the same file? I think this is possible for the gateway, or with customized destination rules. Add @JimmyCYJ to comment.

Reply:
1. Done
2. That is a great point. DRs will have definitely that case. I handled that. PTAL